.github/workflows/protobuf.yaml

```diff
@@ -0,0 +1,47 @@
+name: protobuf
+permissions: read-all
+
+on:
+  pull_request:
+    branches:
+    paths:
+      - '.github/workflows/protobuf.yaml'
+      - 'protobuf/**'
+  push:
+    branches: [ master ]
+    paths:
+      - '.github/workflows/protobuf.yaml'
+      - 'protobuf/**'
+  schedule:
+    - cron: '0 0 * * 0' # weekly
+
+jobs:
+  build:
```

Comment:
The job name is "build" but it builds, checks formatting, analyzes, tests. If we want all of these in a single job maybe a more generic name like "Check protobuf" would be good. (but I think we should split this into jobs, see below)